scipy/optimize/_lsq/least_squares.py

```diff
@@ -258,10 +258,10 @@ def least_squares(
         ``fun(x, *args, **kwargs)``, i.e., the minimization proceeds with
         respect to its first argument. The argument ``x`` passed to this
         function is an ndarray of shape (n,) (never a scalar, even for n=1).
-        It must return a 1-D array_like of shape (m,) or a scalar. If the
-        argument ``x`` is complex or the function ``fun`` returns complex
-        residuals, it must be wrapped in a real function of real arguments,
-        as shown at the end of the Examples section.
+        It must always allocate and return a 1-D array_like of shape (m,) or
```

Comment:
I think that "always" is not really necessary here, do you agree?

Reply:
I agree. I fixed.